x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/math/AutoBucket.java

```diff
@@ -216,6 +216,16 @@ private TypeResolution resolveType(BiFunction<Expression, TypeResolutions.ParamO
         return isFoldable(to, sourceText(), FOURTH);
     }
 
+    private long convertToLong(Expression e, String operationName, TypeResolutions.ParamOrdinal paramOrd) {
+        long v = 0;
+        if (e.dataType() == DataTypes.DATETIME) {
+            v = ((Number) e.fold()).longValue();
+        } else {
+            v = DateFieldMapper.DEFAULT_DATE_TIME_FORMATTER.parseMillis(((BytesRef) e.fold()).utf8ToString());
+        }
+        return v;
+    }
```

Comment:
Only expression e is needed, the other params are not.
The method itself can also be tighten using the ternary operators and existing utilities:
```
val value = Foldables.valueOf(e);
return DataTypes.isDate(e) ? ((Number) value).longValue() : DateFieldMapper.DEFAULT_DATE_TIME_FORMATTER.parseMillis(BytesRefs.toString(value));
```